core/src/main/java/org/elasticsearch/monitor/fs/FsInfo.java

```diff
@@ -496,11 +496,23 @@ public DiskUsage getMostDiskEstimate() {
     private Path total() {
         Path res = new Path();
         Set<String> seenDevices = new HashSet<>(paths.length);
+        Set<String> seenMounts = new HashSet<>(paths.length);
+        boolean seenDevice;
+        boolean seenMount;
         for (Path subPath : paths) {
             if (subPath.path != null) {
-                if (!seenDevices.add(subPath.path)) {
-                    continue; // already added numbers for this device;
-                }
+                seenDevice = !seenDevices.add(subPath.path);
+            } else {
+                seenDevice = true;
+            }
+            if (subPath.mount != null) {
+                seenMount = !seenMounts.add(subPath.mount);
+            } else {
+                seenMount = true;
+            }
+
+            if (seenMount || (seenDevice && seenMount)) {
```

Comment:
This condition is equivalent to simply `seenMount`. Effectively, `seenDevice` is unused. I think this means something is missing/wrong in the test cases.